### ISSUE TITLE
Error when using pilfer via console on Debian Jessie

### ISSUE THREAD

**Issue by peterpan192** [NapoleonWils0n/pilfer#3]
Hey there,
I just installed the new scripts on both my ubuntu-Laptop and on my Raspberry Pi (OSMC - Debian Jessie). It works just fine on ubuntu; on OSMC I get this error, however when I'm trying to use pilfer via the command line:

Traceback (most recent call last):
  File "/home/osmc/.local/bin/pilfer", line 11, in <module>
    load_entry_point('pilfer==2.0', 'console_scripts', 'pilfer')()
  File "/home/osmc/.local/lib/python3.4/site-packages/pilfer/pilfer.py", line 147, in entry
    ffrec = record.ffmpeg(**ffmpegDict)
  File "/home/osmc/.local/lib/python3.4/site-packages/pilfer/record.py", line 64, in ffmpeg
    process = subprocess.run(ffsplit)
AttributeError: 'module' object has no attribute 'run'

Do you have any suggestions what to do?
Thanks a lot!

**Comment by NapoleonWils0n** [comment]
hi mate

the problem is you are running python 3.4 on osmc
pilfer requires python 3.6

see if you can install python 3.6 on osmc with apt
you may need to enable backports to get a newer version of python

**Comment by peterpan192** [comment]
I found out that it's almost impossible (for me) to install a newer version of python in debian jessie. Luckily the new OSMC is based on Debian Stretch where Python 3.5.3 is the most recent stable version. I checked the version of my Ubuntu and it is 3.5.3 as well so i figured that it should also work in Debian Stretch. However, it doesn't. I've been trying for the last two hours. Everything is installed just fine, however nothing happens if I try to record from the command line. It says: 'running ffmpeg command:' but then nothing happens and I can type within the command line a few seconds later. When I'm trying to save a video-url to my Desktop, the Kodi log gives me the following:

Process: Args   : "%s\n" "{0}" > "$HOME/Desktop/url-$(date +"%Y-%m-%d-%H-%M-%S").txt"
02:34:50.195 T:1371714304  NOTICE: Process: Start
02:34:50.345 T:1371714304  NOTICE: ExecuteAppLinux: "printf" "%s\n" "http://arteptweb.gl-systemhaus.de/am/ptweb/042000/042500/042562-000-A_SQ_$
02:34:50.379 T:1371714304  NOTICE: ExecuteAppLinux: Failure: 512
02:34:50.379 T:1371714304  NOTICE: Process: Stop
02:34:50.379 T:1371714304  NOTICE: Process: Showing Kodi window

There is no text file on my Desktop. Do you still think, I have to upgrade to python 3.6 or might there be another problem, since I am not running python 3.6 on Ubuntu and everything works fine there.
Thanks again! 
I really appreciate your help and your reliability!

**Comment by NapoleonWils0n** [comment]
hi mate

sorry your having some bother getting set up on osmc

its really strange you cant save the url from kodi
because saving the url just uses bash to save the url to the desktop and not python

ive never had the save url command not work, and its the same bash code to save the url
as the kodi playercore factory version

the ExecuteAppLinux failure 512 seems to suggest there is some problem launching external players
especially since you cant save the url

can you play videos from kodi using the play menu to open the video with mpv

maybe try reinstalling kodi, 
what version of kodi are you running

it seems pilfer is installed otherwise you wouldnt get the running ffmpeg command text in the terminal

other things that might cause some problems

you need ffmpeg version 3.0 and above

in the pilfer script i check for different operating system like linux windows mac freebsd
im just wondering if osmc doesnt show up as linux

things to check

* check your ffmpeg version by typing ffmpeg in the terminal

is ffmpeg in your bash path

```
ffmpeg
```
* try typing pilfr in the terminal and it should display the pilfer help screen

```
pilfer
```

* check what your operating is by running uname -a in the terminal

```
uname -a
```

there seems to be 2 problems

using the menu from kodi to save urls and record
and running pilfer on the command line and ffmpeg not running

its seems like the script cant find the ffmpeg binary as though its not in your bash path

you can see where ffmpeg is installed by running whereis ffmpeg in the terminal

```
whereis ffmpeg
```

you can check your bash path to see if the directory where ffmpeg is installed is in your bash path by running echo $PATH in the terminal

```
echo $PATH
```

see if any of that helps, but i think it may be more than just python by the sounds of it

**Comment by peterpan192** [comment]
>  what version of kodi are you running

I'm running Kodi 17.6
> check your ffmpeg version by typing ffmpeg in the terminal

ffmpeg version is 3.something
> is ffmpeg in your bash path

Yes, it is.
> try typing pilfr in the terminal and it should display the pilfer help screen

Yes, it does.
> uname -a

Linux osmc 4.9.29-13-osmc #1 SMP PREEMPT Thu Dec 21 17:18:07 UTC 2017 armv7l GNU/Linux

Nevertheless, I figured it out. Since OSMC doesn't come with the folder 'Desktop' ,the first thing I did was to create it. It didn't work until I made the folder writable, readable and executable by anybody with 

`sudo chmod 777 'Desktop'` 

That solved both the problems for me. Now video urls are stored to the Desktop and also pilfer works via the command line which makes almost perfect sense to me. It is clear that it solved the problem of not writing data to the Desktop folder but how did it affect the pilfer script if being started from the command line?

Thanks for you help! 
I am very happy that everything works now!
